test/augmentation/test_torchaudio.py

```diff
@@ -113,6 +143,17 @@ def test_serialize_deserialize_transform_volume(audio):
     assert_array_almost_equal(perturbed_volume, audio * 0.5)
 
 
+@pytest.mark.parametrize("recording_to_dict", [True, False])
+def test_serialize_deserialize_transform_reverb(audio, rir, recording_to_dict):
+    rir = rir.to_dict() if recording_to_dict else rir
+    reverb_orig = ReverbWithImpulseResponse(rir=rir)
+    data_reverb = reverb_orig.to_dict()
+    reverb = ReverbWithImpulseResponse.from_dict(data_reverb)
+    rvb_audio = reverb(audio, SAMPLING_RATE)
```

Comment:
I think you don't need to apply the effect here, just compare equality of transform objects before and after serialization (since they're dataclasses I think __eq__ is auto-defined).